src/MoBi.Presentation/MenusAndBars/ContextMenus/ContextMenuForModule.cs

```diff
@@ -46,6 +48,29 @@ public IContextMenu InitializeWith(ObjectBaseDTO dto, IPresenter presenter)
          return this;
       }
 
+      private IMenuBarItem createDefaultMergeBehaviorItemFor(Module module)
+      {
+         return CreateSubMenu.WithCaption(AppConstants.MenuNames.DefaultMergeBehavior)
+            .WithItem(CreateMenuCheckButton.WithCaption(MergeBehavior.Overwrite.ToString())
+               .WithChecked(module.DefaultMergeBehavior == MergeBehavior.Overwrite)
+               .WithCheckedAction(defaultOverwrite => makeModuleOverwrite(module, defaultOverwrite)))
+            .WithItem(CreateMenuCheckButton.WithCaption(MergeBehavior.Extend.ToString())
+               .WithChecked(module.DefaultMergeBehavior == MergeBehavior.Extend)
+               .WithCheckedAction(defaultExtend => makeModuleExtend(module, defaultExtend)));
+      }
+
+      private void makeModuleExtend(Module module, bool defaultExtend)
+      {
+         if (defaultExtend && module.DefaultMergeBehavior != MergeBehavior.Extend)
+            _container.Resolve<MakeExtendModuleUICommand>().For(module).ExecuteWithinExceptionHandler();
```

Comment:
Why do this look all so weird?

Reply:
It's because the MenuBarCheckItem overrides the default Click handler from MenuBarItem.

Do you think we should just change it in Core? We probably just need to check if there's a command registered and if so, call it. We wouldn't want to call it unless there's a real command though because all the spots where this is used up to now do not use it with a command, so they will have the `NotImplementedCommand`

```
   public class MenuBarCheckButton : MenuBarButton, IMenuBarCheckItem
   {
      private bool _checked;
      public event Action<bool> CheckedChanged = delegate { };

      public bool Checked
      {
         set { _checked = value; CheckedChanged(value); }
         get { return _checked; }
      }

      public override void Click()
      {
         Checked = !Checked;
      }
   }

   public class MenuBarButton : MenuBarItem, IMenuBarButton
   {
      public IUICommand Command { get; set; }

      public MenuBarButton()
      {
         Command = new NotImplementedCommand();
      }

      public virtual void Click()
      {
         Command.ExecuteWithinExceptionHandler();
      }
   }
```